app/models/riiif/image.rb

```diff
@@ -52,9 +52,17 @@ def image
     ##
     # @param [ActiveSupport::HashWithIndifferentAccess] args
     def render(args)
-      options = decode_options!(args)
-      cache.fetch(Image.cache_key(id, options), compress: true, expires_in: Image.expires_in) do
-        image.extract(options)
+      cache_opts = args.select { |a| %w[region size quality rotation format].include? a.to_s }
+      key = Image.cache_key(id, cache_opts)
+
+      file = Pathname.new(cache.send(:normalize_key,
+                                     key,
+                                     compress: true,
+                                     expires_in: Image.expires_in))
+      logger.debug "Cache #{file.exist? ? 'hit' : 'miss'} for #{file}"
```

Comment:
I'll remove this debugging statement if you want, but I've found it extremely useful.